gateway-ha/src/test/java/io/trino/gateway/ha/HaGatewayTestUtils.java

```diff
@@ -159,6 +142,31 @@ public static void setUpBackend(
                         .build();
         Response response = httpClient.newCall(request).execute();
         assertThat(response.isSuccessful()).isTrue();
+        TrinoHealthStateType newClusterHealthState = TrinoHealthStateType.PENDING;
+        long startTime = System.currentTimeMillis();
+        long lastExecutionTime = System.currentTimeMillis();
+        // pull cluster health states for 10 seconds
+        // It should be enough as the healthcheck is run every second
+        int timeout = 10 * 1000;
+        while (newClusterHealthState != TrinoHealthStateType.HEALTHY && (lastExecutionTime - startTime) < timeout) {
+            // check the state of newly added cluster every second
+            if (System.currentTimeMillis() - lastExecutionTime <= 1000) {
+                continue;
```

Comment:
question: should we be adding some form of yield here so the while loop isn't burning through CPU cycles? maybe a 100ms sleep or something

Reply:
as in `Thread.sleep(100);`? In that case do we still need this while loop?

Reply:
+1 you should refactor more like:
```java
        int timeout = 10 * 1000;
        while (newClusterHealthState != TrinoHealthStateType.HEALTHY && (System.currentTimeMillis() - startTime) < timeout) {
          // do whatever logic
          Thread.sleep(100);
        }
```

Reply:
you'd still need the while loop right, as you want to check the cluster healthy every 1 second or so?

you could update this to be:
```
            if (System.currentTimeMillis() - lastExecutionTime <= 1000) {
                Thread.sleep(System.currentTimeMillis() - lastExecutionTime)
                continue;
            }
```

Or later on for the `if (response.isSuccessful()) {` call you could have an else Thread.sleep(1000) and remove this current if-condition